app/components/Reactions/ReactionList.tsx

```diff
@@ -0,0 +1,82 @@
+import { observer } from "mobx-react";
+import React from "react";
+import type { Reaction as TReaction } from "@shared/types";
+import useHover from "~/hooks/useHover";
+import { EmojiReactedUsers, ReactionData } from "~/types";
+import Logger from "~/utils/Logger";
+import Flex from "../Flex";
+import Reaction from "./Reaction";
+
+type Props = {
+  /** List of active reactions. */
+  reactions: TReaction[];
+  /** Callback when the user intends to add a reaction. */
+  onAddReaction: (emoji: string) => Promise<void>;
+  /** Callback when the user intends to remove a reaction. */
+  onRemoveReaction: (emoji: string) => Promise<void>;
+  /** Callback to fetch the detailed reaction data (incl. users' info) - needed for the tooltip. */
+  fetchReactionData: () => Promise<ReactionData[]>;
+  /** classname generated by styled-components */
+  className?: string;
+};
+
+const ReactionList: React.FC<Props> = ({
+  reactions,
+  onAddReaction,
+  onRemoveReaction,
+  fetchReactionData,
+  className,
+}) => {
+  const [emojiReactedUsers, setEmojiReactedUsers] =
+    React.useState<EmojiReactedUsers>({});
+  const { hovered, onMouseEnter, onMouseLeave } = useHover({
+    duration: 250,
+  });
+
+  React.useEffect(() => {
+    const loadReactionData = async () => {
+      try {
+        const reactionData = await fetchReactionData();
+
+        const transformedData = reactionData.reduce((acc, data) => {
```

Comment:
This exact logic is in two spots, can we make it a `@computed` getter on the comment model?

Reply:
Current code was doing unnecessary transformations. On refactoring, this data has become an `observable` (we need it to render the tooltip for reactions added through websockets).